http/sys_version_history_test.go

```diff
@@ -0,0 +1,81 @@
+package http
+
+import (
+	"testing"
+
+	"github.com/hashicorp/vault/sdk/version"
+	"github.com/hashicorp/vault/vault"
+)
+
+// TestSysVersionHistory_List tests the sys/version-history endpoint. Requests
+// to the endpoint must be authenticated. Without synthetically altering the
+// underlying core/versions storage entries, a single version entry should
+// exist.
+func TestSysVersionHistory_List(t *testing.T) {
```

Comment:
Typically we put only tests in the `http` package when they're exercising code that lives in that package, i.e. "non-logical" endpoints.

Reply:
I was trying to follow the example of testing other sys endpoints:

```
❯ find http -type f -name sys*_test.go
http/sys_rekey_test.go
http/sys_init_test.go
http/sys_version_history_test.go
http/sys_rotate_test.go
http/sys_generate_root_test.go
http/sys_in_flight_requests_test.go
http/sys_lease_test.go
http/sys_policy_test.go
http/sys_monitor_test.go
http/sys_auth_test.go
http/sys_mounts_test.go
http/sys_config_cors_test.go
http/sys_hostinfo_test.go
http/sys_health_test.go
http/sys_metrics_test.go
http/sys_mount_test.go
http/sys_internal_test.go
http/sys_seal_test.go
http/sys_audit_test.go
http/sys_config_state_test.go
http/sys_leader_test.go
http/sys_wrapping_test.go
```

Reply:
Some of these are indeed non-logical endpoints, e.g. health, metrics, pprof, monitor.  Most of the rest are just ancient tests that we probably wouldn't write this way nowadays.  If you look for tests starting with "TestSystemBackend" you'll find a ton of other examples elsewhere that are more conventional.

Reply:
That's great context, thank you! I'll take a look and modify accordingly.